backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java

```diff
@@ -317,8 +321,17 @@ public FutureExperimentRunDAO(
                 .execute());
   }
 
-  private InternalFuture<Void> checkProjectPermission(
-      List<String> projId, ModelDBActionEnum.ModelDBServiceActions action) {
+  public interface CheckPermissionBasedOnResourceTypesFunction {
+    InternalFuture<Void> checkPermission(
+        List<String> projId,
+        ModelDBActionEnum.ModelDBServiceActions action,
+        ModelDBServiceResourceTypes modelDBServiceResourceTypes);
+  }
+
+  private InternalFuture<Void> checkEntityPermissionBasedOnResourceTypes(
```

Comment:
I'm confused between `checkPermission` defined before and this. They have the same signature but somehow they're different.

Reply:
Above `checkPermission`  was interface method so while we are passing function as an argument in handler we have to implement interface method so that method will be `checkEntityPermissionBasedOnResourceTypes` with same argument. SO finally `checkEntityPermissionBasedOnResourceTypes` method and new introduced interface works for the function as argument in handler

Reply:
Rename interface name from `checkPermission` to `checkEntityPermissionBasedOnResourceTypes` to reduce confusion